e2e/tests/test_e2e.py

```diff
@@ -1419,52 +1421,79 @@ def test_zzzz_cluster_deletion(self):
         k8s.update_config(patch_delete_annotations)
 
     @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
-    def test_decrease_max_connections(self):
+    def test_patroni_config_update(self):
         '''
-            Test decreasing max_connections and restarting cluster through rest api
+            Change Postgres config under Spec.Postgresql.Parameters and Spec.Patroni
+            and query Patroni config endpoint to check if manifest changes got applied
+            via restarting cluster through Patroni's rest api
         '''
         k8s = self.k8s
-        cluster_label = 'application=spilo,cluster-name=acid-minimal-cluster'
-        labels = 'spilo-role=master,' + cluster_label
-        new_max_connections_value = "99"
-        pods = k8s.api.core_v1.list_namespaced_pod(
-            'default', label_selector=labels).items
-        self.assert_master_is_unique()
-        masterPod = pods[0]
+        masterPod = k8s.get_cluster_leader_pod()
+        labels = 'application=spilo,cluster-name=acid-minimal-cluster,spilo-role=master'
         creationTimestamp = masterPod.metadata.creation_timestamp
 
         # adjust max_connection
-        pg_patch_max_connections = {
+        pg_patch_config = {
             "spec": {
                 "postgresql": {
                     "parameters": {
-                        "max_connections": new_max_connections_value
+                        "max_connections": "50"
                      }
+                 },
+                 "patroni": {
+                    "slots": {
+                        "test_slot": {
+                            "type": "physical"
+                        }
+                    },
+                    "ttl": 29,
+                    "loop_wait": 9,
+                    "retry_timeout": 9,
+                    "synchronous_mode": True
                  }
             }
         }
-        k8s.api.custom_objects_api.patch_namespaced_custom_object(
-            "acid.zalan.do", "v1", "default", "postgresqls", "acid-minimal-cluster", pg_patch_max_connections)
 
-        def get_max_connections():
+        try:
+            k8s.api.custom_objects_api.patch_namespaced_custom_object(
+                "acid.zalan.do", "v1", "default", "postgresqls", "acid-minimal-cluster", pg_patch_config)
+            
+            self.eventuallyEqual(lambda: self.k8s.get_operator_state(), {"0": "idle"}, "Operator does not get in sync")
+
+            def compare_config():
+                effective_config = k8s.patroni_rest(masterPod.metadata.name, "config")
```

Comment:
but that only tells you if the Patroni config was updated, not if the change [was actually applied on the PG level](https://github.com/zalando/postgres-operator/pull/1535/files#diff-0b56893556c9ee8b6a3f37f0e8984f343da0fd397154b370938a433fa4aa475dR1427-R1428).
Patroni has[ the `pending_restart` flag](https://github.com/zalando/patroni/blob/75e52226a848d64317ac98264234e62706fd762c/docs/rest_api.rst) to indicate such situation
```bash
$ curl -s http://localhost:8008/patroni | jq .
{
  "pending_restart": true,
```

Reply:
true. Added a check to query the settings in the database